python/mxnet/ndarray/contrib.py

```diff
@@ -542,14 +545,17 @@ def isnan(data):
     [1. 0.]
     <NDArray 2 @cpu(0)>
     """
-    return data != data # pylint: disable=comparison-with-itself
+    return data != data  # pylint: disable=comparison-with-itself
 
-def adamw_update(weight, grad, mean, var, rescale_grad, lr, eta, beta1=0.9, beta2=0.999,
-                 epsilon=1e-8, wd=0, clip_gradient=-1, out=None, name=None, **kwargs):
+def getRescaleGrad(rescale_grad, ctx):
```

Comment:
Maybe check whether rescale_gradis already on the proper context and skip the copy if it is?

Reply:
Copy is always skipped when `rescale_grad` is defined in a proper context:
```
    def as_in_context(self, context):
        if self.context == context:
            return self
        return self.copyto(context)
```

Reply:
Oh, that's good to know. Thanks :-)